### ISSUE TITLE
Add support for SNO (Single Node)

### ISSUE THREAD

**Issue by rbo** [RedHat-EMEA-SSA-Team/hetzner-ocp4#176]
Please add SNO support, cluster.yaml:

```
compute_count: 0
master_count: 1
```

Bootstrap in place wont work, because in place bootstrap might work only if you boot from iso and not direct into qcow2.

**Comment by rbo** [comment]
Installation still work-in-progress, but looks not so bad:
```bash
$ cat cluster-sno.yaml
cluster_name: sno
sdn_plugin_name: OVNKubernetes
vn_subnet: "192.168.56.0"

openshift_mirror: "https://mirror.openshift.com"

openshift_version: 4.8.0-fc.7
openshift_location: "{{ openshift_mirror }}/pub/openshift-v4/clients/ocp-dev-preview/{{openshift_client_version}}"

coreos_version: 4.8.0-fc.7
coreos_download_url: "{{ openshift_mirror }}/pub/openshift-v4/dependencies/rhcos/pre-release/{{ coreos_version }}/rhcos-{{coreos_version}}-x86_64-qemu.x86_64.qcow2.gz"
coreos_csum_url: "{{ openshift_mirror }}/pub/openshift-v4/dependencies/rhcos/pre-release/{{ coreos_version }}/sha256sum.txt"

#post_install_add_ons:
#  - name: 'ntp'
#    tasks_from: 'post-install.yml'

compute_count: 0
master_count: 1
$ oc get etcd/cluster -o yaml | grep -C5 single
kind: Etcd
metadata:
  annotations:
    include.release.openshift.io/ibm-cloud-managed: "true"
    include.release.openshift.io/self-managed-high-availability: "true"
    include.release.openshift.io/single-node-developer: "true"
    release.openshift.io/create-only: "true"
  creationTimestamp: "2021-06-07T18:40:26Z"
  generation: 1
  name: cluster
  resourceVersion: "19598"
$ oc get clusterversion
NAME      VERSION   AVAILABLE   PROGRESSING   SINCE   STATUS
version             False       True          23m     Unable to apply 4.8.0-fc.7: the cluster operator monitoring has not yet successfully rolled out
```

**Comment by rbo** [comment]
Installation done, works fine. So SNO works very well with simple cluster.yml settings:
```
compute_count: 0
master_count: 1
```

**Comment by rbo** [comment]
Everything is documented in devel branch, will be merged into master with PR #183
